GTMAppAuthSwift/Sources/GTMAppAuthFetcherAuthorization.swift

```diff
@@ -24,6 +24,192 @@ import GTMSessionFetcherCore
+    }
+    return authorization
+  }
+
+  // MARK: - Removing Authorizations
+
+  /// Removes the saved authorization for the supplied name.
+  ///
+  /// - Parameter itemName: The `String` name for the authorization saved in the keychain.
+  /// - Throws: Any error that may arise during removal, including `KeychainWrapper.Error`.
+  @objc public final class func removeAuthorization(for itemName: String) throws {
+    let keychain = keychain ?? GTMKeychain()
+    try keychain.removePasswordFromKeychain(forName: itemName)
+  }
+
+  /// Removes the saved authorization for the supplied name.
+  ///
+  /// - Parameters:()
+  ///   - itemName: The `String` name for the authorization saved in the keychain.
+  ///   - usingDataProtectionKeychain: A `Bool` detailing whether or not to use the data protection
+  ///     keychain.
+  /// - Throws: Any error that may arise during removal, including `KeychainWrapper.Error`.
+  @available(macOS 10.15, *)
+  @objc public final class func removeAuthorization(
+    for itemName: String,
+    usingDataProtectionKeychain: Bool
+  ) throws {
+    let keychain = keychain ?? GTMKeychain()
+    try keychain.removePasswordFromKeychain(
+      forName: itemName,
+      usingDataProtectionKeychain: usingDataProtectionKeychain
+    )
+  }
+
+  // MARK: - Saving Authorizations
+
+  /// Saves the passed authorization with the provided name.
+  ///
+  /// - Parameters:
+  ///   - authorization: An instance of `GMTAppAuthFetcherAuthorization`.
+  ///   - itemName: The `String` name for the authorization to save in the Keychain.
+  /// - Throws: Any error that may arise during removal, including `KeychainWrapper.Error`.
+  @objc public final class func save(
+    authorization: GTMAppAuthFetcherAuthorization,
+    with itemName: String
+  ) throws {
+    let keychain = keychain ?? GTMKeychain()
+    if #available(macOS 10.13, iOS 11, tvOS 11, *) {
+      try modernArchive(authorization: authorization, itemName: itemName)
+    } else {
+      let authorizationData = NSKeyedArchiver.archivedData(withRootObject: authorization)
+      try keychain.save(passwordData: authorizationData, forName: itemName)
+    }
+  }
+
+  /// Saves the passed authorization with the provided name.
+  ///
+  /// - Parameters:
+  ///   - authorization: An instance of `GMTAppAuthFetcherAuthorization`.
+  ///   - itemName: The `String` name for the authorization to save in the Keychain.
+  ///   - usingDataProtectionKeychain: A `Bool` detailing whether or not to use the data protection
+  ///     keychain.
+  /// - Throws: Any error that may arise during removal, including `KeychainWrapper.Error`.
+  @available(macOS 10.15, *)
+  @objc public final class func save(
+    authorization: GTMAppAuthFetcherAuthorization,
+    with itemName: String,
+    usingDataProtectionKeychain: Bool
+  ) throws {
+    let keychain = keychain ?? GTMKeychain()
+    // Don't need to check macOS here because method is only available for 10.15 and higher.
```

Comment:
My understanding of the `@available` is that it is not exclusive. The `*` operates as a kind of wildcard operator; i.e., it matches everything not otherwise listed prior to it.

https://docs.swift.org/swift-book/ReferenceManual/Attributes.html
https://nshipster.com/available/#shorthand-specification

So, for example, if I have a cross platform app and have this type:

```
struct Foo {
  @available(iOS 13, *)
  func bar() -> String {
    return "BAR!"
  }
}
```

The method `Foo.bar()` is limited on iOS to v13 or higher, but doesn't say anything about macOS. This doesn't mean it isn't callable from macOS. It just means that its usage is not restricted to a certain version of macOS. Feel free to experiment locally.

If I'm correct on the above, then this code limits its availability on macOS to 10.15 and higher, and then the rest is limited by the Swift package's min versions. Within the method's implementation, we fork our logic based on whether or not we are running on iOS 11 or tvOS 11. 

Make sense? Or, perhaps my thinking is off. LMK.

Reply:
No, I think this is my mistake. I've updated things on the archiving and unarchiving. Let me know if this looks better.